workbench-core/example/app/src/routes/datasetRoutes.ts

```diff
@@ -73,14 +75,43 @@ export function setUpDSRoutes(
     })
   );
 
+  // unshare dataset
+  router.delete(
+    '/datasets/:datasetId/share/:endpointId',
+    wrapAsync(async (req: Request, res: Response) => {
+      if (
+        req.params.datasetId.match(uuidWithLowercasePrefixRegExp(dataSetPrefix)) === null ||
+        req.params.endpointId.match(uuidWithLowercasePrefixRegExp(endPointPrefix)) === null
+      ) {
+        throw Boom.badRequest('datasetId and endpointId parameters must be valid');
+      }
+
+      await dataSetService.removeDataSetExternalEndpoint(
+        req.params.datasetId,
+        req.params.endpointId,
+        dataSetStoragePlugin
+      );
+      res.status(204).send();
+    })
+  );
+
+  // List storag locations
+  router.get(
+    '/datasets/storage',
+    wrapAsync(async (req: Request, res: Response) => {
+      const locations = await dataSetService.listStorageLocations();
+      res.send(locations);
+    })
+  );
```

Comment:
Is this from your listStorageLocations integ tests work?

Reply:
again, yes